driver-core/src/main/com/mongodb/client/model/GeoNearOption.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model;
+
+import org.bson.Document;
+
+/**
+ * Optional fields for the {@link Aggregates#geoNear} pipeline stage.
+ *
+ * @since 4.8
+ */
+public final class GeoNearOption {
```

Comment:
This seems like a new pattern for Aggregates builder methods.  Looking at what we have currently, there is BucketOptions, BucketAutoOptions, GraphLookupOptions, UnwindOptions, MergeOptions, etc.  all of which have a different design.

What's the rationale for diverging here?

Reply:
Updated